packages/flutter_tools/test/general.shard/github_template_test.dart

```diff
@@ -33,45 +42,89 @@ void main() {
-          'A+crash'
-      );
-    }, overrides: <Type, Generator>{
-      HttpClientFactory: () => () => FakeHttpClient(),
+      setUp(() async {
+        stackTrace = StackTrace.fromString('trace');
+      });
+
+      testUsingContext('shortened', () async {
+        final GitHubTemplateCreator creator = GitHubTemplateCreator();
+        expect(
+            await creator.toolCrashIssueTemplateGitHubURL(command, errorString, exception, stackTrace, doctorText),
+            _kShortURL
+        );
+      }, overrides: <Type, Generator>{
+        HttpClientFactory: () => () => SuccessShortenURLFakeHttpClient(),
+        FileSystem: () => MemoryFileSystem(),
+        ProcessManager: () => FakeProcessManager.any(),
+      });
+
+      testUsingContext('with network failure', () async {
+        final GitHubTemplateCreator creator = GitHubTemplateCreator();
+        expect(
+            await creator.toolCrashIssueTemplateGitHubURL(command, errorString, exception, stackTrace, doctorText),
+            'https://github.com/flutter/flutter/issues/new?title=%5Btool_crash%5D+this+is+a+100%25+error&body=%23%'
+                '23+Command%0A++%60%60%60%0A++flutter+test%0A++%60%60%60%0A%0A++%23%23+Steps+to+Reproduce%0A++1.+...'
+                '%0A++2.+...%0A++3.+...%0A%0A++%23%23+Logs%0A++failing+to+succeed%21%21%21%0A++%60%60%60%0A++trace%0A'
+                '++%60%60%60%0A++%60%60%60%0A+++%5B%E2%9C%93%5D+Flutter+%28Channel+report%0A++%60%60%60%0A%0A++%23%23'
+                '+Flutter+Application+Metadata%0A++No+pubspec+in+working+directory.%0A++&labels=tool%2Csevere%3A+crash'
+        );
+      }, overrides: <Type, Generator>{
+        HttpClientFactory: () => () => FakeHttpClient(),
+        FileSystem: () => MemoryFileSystem(),
+        ProcessManager: () => FakeProcessManager.any(),
+      });
+
+      testUsingContext('app metadata', () async {
+        final GitHubTemplateCreator creator = GitHubTemplateCreator();
+        final Directory projectDirectory = fs.currentDirectory;
+        final File pluginsFile = projectDirectory.childFile('.flutter-plugins');
+
+        projectDirectory
+            .childFile('pubspec.yaml')
+            .writeAsStringSync('''
+name: failing_app
+version: 2.0.1+100
+flutter:
+  uses-material-design: true
+  module:
+    androidX: true
+    androidPackage: com.example.failing.android
+    iosBundleIdentifier: com.example.failing.ios
+''');
+
+        pluginsFile
+            .writeAsStringSync('''
+camera=/fake/pub.dartlang.org/camera-0.5.7+2/
+device_info=/fake/pub.dartlang.org/pub.dartlang.org/device_info-0.4.1+4/
+        ''');
+
+        final String actualURL = await creator.toolCrashIssueTemplateGitHubURL(command, errorString, exception, stackTrace, doctorText);
+        final String body = Uri.parse(actualURL).queryParameters['body'];
+        expect(body, contains('**Version**: 2.0.1+100\n'));
```

Comment:
Any reason to just not assert the exact string?  It seems like this could end up giving us false positives

Reply:
I did it for readability, but that wasn't necessary once I decided to decode it.  Will do.